src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrder.java

```diff
@@ -0,0 +1,165 @@
+                .getTypeParameters(ast);
+
+        final FileContents contents = getFileContents();
+        final int lineNo = ast.getLineNo();
+        final TextBlock textBlock = contents.getJavadocBefore(lineNo);
+
+        final List<JavadocTag> javadocParams = getJavadocTags(textBlock);
+
+        method1(methodParams, methodTypeParams, javadocParams);
+    }
+
+    /**
+     * Checks a set of tags for matching parameters.
+     *
+     *  @param methodParams list of method parameters
+     * @param params list of method type parameters
+     * @param javadocParams list of javadoc parameters
+     */
+    private void method1(final List<DetailAST> methodParams,
+                                final List<DetailAST> params,
+                         final List<JavadocTag> javadocParams) {
+
+        final int totalMethodParamSize = methodParams.size() + params.size();
+        //Merge both lists
+        params.addAll(methodParams);
+
+        final ListIterator<JavadocTag> javadocTag = javadocParams.listIterator();
+        final ListIterator<DetailAST> methodTag = params.listIterator();
+        while (javadocTag.hasNext() && methodTag.hasNext()) {
+            final JavadocTag tag = javadocTag.next();
+            final DetailAST mTag = methodTag.next();
+
+            if (!tag.isParamTag()) {
+                continue;
+            }
+
+            javadocTag.remove();
+            methodTag.remove();
+            final String arg1 = tag.getFirstArg();
+            final String arg2 = mTag.getText();
+
+            if(!arg1.equals(arg2)) {
+                log(tag.getLineNo(), tag.getColumnNo(), MSG_JAVADOC_TAG_ORDER,
+                        "@param", arg1);
+            }
+        }
+    }
+
+    /**
+     * Computes the parameter nodes for a method.
+     *
+     * @param ast the method node.
+     * @return the list of parameter nodes for ast.
+     */
+    private static List<DetailAST> getParameters(DetailAST ast) {
+        final DetailAST params = ast.findFirstToken(TokenTypes.PARAMETERS);
+        final List<DetailAST> returnValue = new ArrayList<>();
+
+        DetailAST child = params.getFirstChild();
+        while (child != null) {
+            if (child.getType() == TokenTypes.PARAMETER_DEF) {
+                final DetailAST ident = child.findFirstToken(TokenTypes.IDENT);
+                if (ident != null) {
+                    returnValue.add(ident);
+                }
+            }
+            child = child.getNextSibling();
+        }
+        return returnValue;
+    }
+
+    /**
+     * Gets all standalone tags from a given javadoc.
+     *
+     * @param textBlock the Javadoc comment to process.
+     * @return all standalone tags from the given javadoc.
+     */
+    private static List<JavadocTag> getJavadocTags(TextBlock textBlock) {
+        final JavadocTags tags = JavadocUtil.getJavadocTags(textBlock,
+            JavadocUtil.JavadocTagType.BLOCK);
```

Comment:
Wouldn't this return the all JavadocTags (like `return`, `throws`, `inline tags`) along with param tag?

Reply:
maybe thats why if condition is there.

Reply:
thats why is condition is there and i made the changes and i will update it along with other
changes are- i moved final `DetailAST mTag = methodTag.next();` below if condition.